101-lab/content/16_templates.md

```diff
@@ -0,0 +1,100 @@
+# Templates
+
+So far in the lab exercises, we've primarily used the 'new-app' command to create many our OpenShift objects. However, all objects can be created and modified using YAML. Templates can be used to rapidly create multiple OpenShift objects from a single YAML file. This can be useful for testing your application in different environments (dev, test, prod) or for recreating similar applications. It is also helpful to use manifests and templates as a strategy for recovering from an incident. 
+
+In this section, we'll create a template that includes all of the objects we've created for our rocketchat application and mongo database. 
+
+You may find it helpful to install a tool such as [Visual Studio Code](https://code.visualstudio.com/) to help work through this section of the lab.  
+
+## Get the YAML
+
+We can use the `oc get` command to gather the YAML files that represent our OpenShift commands and their configuration. We need to collect the YAML for a variety of objects. After replacing [username] we can do this with the following command: 
+
+```
+oc -n [-dev] get deployment,dc,route,service,configmap,pvc,secrets -l app=rocketchat-[username] -o yaml > template.yaml
+```
+The command above will save the YAML definitions for these objects on your local machine in a file named `template.yaml`. 
+
+**Note:** we've also created some objects in the [-tools] namespace including our buildconfig and imagestream, but for the purposes of this demonstration we're not going to add them to our template.
+
+## Remove unnecessary information
+
+Explore the new file that you just created called `template.yaml`
+
+We're now looking to strip unique information generated by OpenShift from the YAML to allow it to be reusable. Search for each of these fields and remove their contents from your `template.yaml` then save it: 
+
+- annotations 
```

Comment:
do we want to share the script that I show you earier in this doc? Ian actually push it to here:https://github.com/BCDevOps/openshift-wiki/blob/master/docs/ArgoCD/get_ns_resources.sh